docs/standards/smart-contracts/lsp7-digital-asset.md

```diff
@@ -265,6 +265,221 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
+function _transfer(
+    address from,
+    address to,
+    uint256 amount,
+    bool force,
+    bytes memory data
+) internal virtual
+```
+
+Transfers `amount` tokens from `from` to `to`.
+
+_Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully transferred._
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `amount` | uint256  | The amount of token to transfer. |
+| `force`  | bool     | When set to TRUE, `to` may be any address; when set to FALSE `to` must be a contract that supports LSP1 UniversalReceiver and not revert. |
+| `memory` | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+:::note
+
+#### Requirements:
+
+- `to` cannot be the zero address.
+- `from` cannot be the zero address.
+- `from` must have at least `amount` tokens.
+- If the caller is not `from`, it must be an operator for `from` with access to at least `amount` tokens.
+
+:::
+
+### _burn
+
+```solidity
+function _burn(
+    address from,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+Burns `amount` tokens.
+
+_Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully transferred._
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `amount` | uint256  | The amount of token to burn. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hook to `from` address. |
+
+:::note
+
+#### Requirements:
+
+- `from` cannot be the zero address.
+- `from` must have at least `amount` tokens.
+- If the caller is not `from`, it must be an operator for `from` with access to at least `amount` tokens.
+
+:::
+
+### _beforeTokenTransfer
+
+```solidity
+function _beforeTokenTransfer(
+    address from,
+    address to,
+    uint256 amount
+) internal virtual
+```
+
+Hook that is called before any token transfer. This includes minting and burning.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `amount` | uint256  | The amount of token to transfer. |
+
+:::note
+
+#### Requirements:
+
+- When `from` and `to` are both non-zero, ``from``'s `amount` tokens will betransferred to `to`.
+- When `from` is zero, `amount` tokens will be minted for `to`.
+- When `to` is zero, ``from``'s `amount` tokens will be burned.
+
+:::
+
+### _notifyTokenSender
+
+```solidity
+function _notifyTokenSender(
+    address from,
+    address to,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token sender about the `amount` tokens changing owners using LSP1 interface.
```

Comment:
An attempt is made to notify the token sender about the `amount` tokens being sent by calling the universalReceiver function on the sender address if it exist.
I guess this sounds better